File: 145. Binary Tree Postorder Traversal/145.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def postorderTraversal(self, root: TreeNode) -> List[int]:
        if root == None:
            return []
        else:
            arr = []
            if root.left:
                arr += self.postorderTraversal(root.left)
            if root.right:
                arr += self.postorderTraversal(root.right)
            
            arr.append(root.val)
            
            return arr

File: sprint/binaryTree/recursionTraversal/145.cpp
#include <vector>
using namespace std;

struct TreeNode {
    int val;
    TreeNode* left;
    TreeNode* right;
};

class Solution {
public:
    vector<int> postorderTraversal(TreeNode* root) {
        vector<int> res;
        traversal(root, res);   
        return res;
    }

    void traversal(TreeNode* root, vector<int>& res) {
        if (root == nullptr) {
            return;
        }
        traversal(root->left, res);
        traversal(root->right, res);
        res.push_back(root->val);
    }
};

int main(void) {
    TreeNode* t = new TreeNode();
}